perfkitbenchmarker/flag_util.py

```diff
@@ -304,3 +306,60 @@ def DEFINE_units(name, default, help, convertible_to=None,
   serializer = UnitsSerializer()
 
   flags.DEFINE(parser, name, default, help, flag_values, serializer, **kwargs)
+
+
+# The YAML flag type is necessary because flags can be read either via
+# the command line or from a config file. If they come from a config
+# file, they will already be parsed as YAML, but if they come from the
+# command line, they will be raw strings. The point of this flag is to
+# guarantee a consistent representation to the rest of the program.
+class YAMLParser(flags.ArgumentParser):
+  """Parse a flag containing YAML."""
+
+  syntactic_help = 'A YAML expression. Ex: "{x: y, foo: 2.3}".'
```

Comment:
Just something to keep in mind: if this is going to end up in --help output, it may not be useful for most cases, where a more specific format of input is expected. You may need to subclass this for specific flags to provide a more specific description, e.g. "an integer or a string", or you may want to make it an instance member and allow it to be overridden in DEFINE_yaml, but if so, none of the more specific checks would happen in the flag parser itself and would need to happen in a config option decoder later.